onnxscript/testing/benchmark/benchmark_helpers.py

```diff
@@ -0,0 +1,682 @@
+):
+    """
+    Exports a model into a folder.
+
+    Args:
+        model: model
+        exporter: script, dynamo
+        folder: folder to export into
+        filename: onnx filename
+        inputs: inputs
+        dynamic_shapes: dynamic shapes
+        target_opset: target opset
+        optimization: optimization scenario
+        verbose: verbosity
+        stats: if not None, populates this
+            dictionary with statistics about time
+
+    Returns:
+        onnx proto
+
+    """
+    import torch.onnx
+
+    if folder:
+        if not os.path.exists(folder):
+            os.mkdir(folder)
+        filename = os.path.join(folder, filename)
+
+    if verbose:
+        print(f"[common_export] start exporting with {exporter!r} in {filename!r}")
+    begin = time.perf_counter()
+    if exporter == "script":
+        torch.onnx.export(
+            model,
+            inputs,
+            filename,
+            do_constant_folding=False,
+            input_names=[f"input{i}" for i in range(len(inputs))],
+            opset_version=target_opset,
+        )
+    elif exporter == "dynamo":
+        with torch.no_grad():
+            prog = torch.onnx.dynamo_export(model, *inputs)
+        onx = prog.model_proto
+        with open(filename, "wb") as f:
+            f.write(onx.SerializeToString())
+    else:
+        raise AssertionError(f"Unknown exporter {exporter!r}")
+
+    if stats is not None:
+        stats["export_time"] = time.perf_counter() - begin
+
+    if verbose:
+        print(f"[common_export] exporter done in {time.perf_counter() - begin}s")
+        print(f"[common_export] size of the export: {os.stat(filename).st_size / 2**20} Mb")
+
+    with open(filename, "rb") as f:
+        onx = onnx.load(filename)
+
+    if optimization:
+        if verbose:
+            print(f"[common_export] start optimization with {optimization!r}")
+        begin = time.perf_counter()
+        onx = optimize_model_proto(onx, optimization, verbose=verbose, stats=stats)
+        end = time.perf_counter() - begin
+        if stats is not None:
+            stats["optimization_time"] = end
+        if verbose:
+            print(f"[common_export] optimization done in {end}")
+            print(f"[common_export] saves the model in {filename!r}")
+            begin = time.perf_counter()
+
+        with open(filename, "wb") as f:
+            f.write(onx.SerializeToString())
+        if verbose:
+            print(f"[common_export] done saving in {time.perf_counter() - begin}")
+
+    return onx
+
+
+def apply_rule_sets(
+    model_proto: onnx.ModelProto,
+    rule_sets: list[str],
+    stats: dict[str, Any] | None = None,
+    verbose: int = 0,
+):
+    """
+    Applies set of patterns on a model to optimizes.
+
+    Args:
+        model_proto: model
+        rule_sets: sets ot apply
+        stats: add statistics if not empty
+        verbose: verbosity
+
+    Returns:
+        optimized model
+    """
+    from onnxscript import ir
```

Comment:
```suggestion
```

can be moved to top because onnxscript is always available for this script.

Reply:
In this file, many imports are like this to reduce the import time. The version of the script was importing onnxruntime and "<script> --help" could take a very long time (> 4 seconds). import onnxscript.ir looks ok. I'll move it.

Reply:
I see. That makes sense. For the onnxruntime import it would be helpful to note this consideration as a comment so future changes won't accidentally move them.

Reply:
done